zowex/src/main.rs

```diff
@@ -261,6 +404,212 @@ fn get_beg(buf: &str) -> Vec<String> {
+    if env::consts::OS == "windows" {
+        zowe_cmd = "zowe.cmd";
+    } else {
+        zowe_cmd = "zowe";
+    }
+
+    // find every program in our path that would execute a 'zowe' command
+    const NOT_FOUND: &str = "notFound";
+    let mut njs_zowe_path: String = NOT_FOUND.to_string();
+    let path = env::var_os("PATH");
+    let path_ext = env::var_os("PATHEXT");
+    for njs_zowe_path_buf in PathSearcher::new(
+        zowe_cmd,
+        path.as_ref().map(OsString::as_os_str),
+        path_ext.as_ref().map(OsString::as_os_str),
+    ) {
+        njs_zowe_path = njs_zowe_path_buf.to_string_lossy().to_string();
+        if njs_zowe_path.to_lowercase().eq(&my_exe_path.to_lowercase()) {
+            // We do not want our own rust executable. Keep searching.
+            njs_zowe_path = NOT_FOUND.to_string();
+            continue;
+        }
+
+        // use the first 'zowe' command on our path that is not our own executable
+        break;
+    }
+    if njs_zowe_path == NOT_FOUND {
+        println!("Could not find a NodeJS zowe command on your path.");
+        println!("Will not be able to run Zowe commands. Terminating.");
+        std::process::exit(EXIT_CODE_NO_NODEJS_ZOWE_ON_PATH);
+    }
+
+    return njs_zowe_path;
+}
+
+/**
+ * Is the zowe daemon currently running?
+ * @returns A structure that indicates if the daemon is running, and if so
+ *          properties about that running process.
+ */
+fn is_daemon_running() -> DaemonProcInfo {
+    let mut sys = System::new_all();
+    sys.refresh_all();
+    for (pid, process) in sys.processes() {
+        if process.name().to_lowercase().contains("node") &&
+           process.cmd()[1].to_lowercase().contains("@zowe") &&
+           process.cmd()[1].to_lowercase().contains("cli") &&
+           process.cmd()[2].to_lowercase() == "--daemon"
+        {
+            // convert the process command from a vector to a string
+            let mut proc_cmd: String = String::new();
+            for cmd_part in process.cmd() {
+                proc_cmd.push_str(cmd_part);
+                proc_cmd.push(' ');
+            }
+            return DaemonProcInfo {
+                is_running: true,
+                name: process.name().to_string(),
+                pid: pid.to_string(),
+                cmd: proc_cmd
+            };
+        }
+    }
+    return DaemonProcInfo {
+        is_running: false,
+        name: "no name".to_string(),
+        pid: "no pid".to_string(),
+        cmd: "no cmd".to_string()
+    };
+}
+
+/**
+ * Should we use daemon mode? The user controls this with an environment variable.
+ * @returns true or false.
+ */
+fn user_wants_daemon() -> bool {
+    const DAEMON_ENV_VAR_NM: &str = "ZOWE_USE_DAEMON";
+    let env_var_val;
+    match env::var(DAEMON_ENV_VAR_NM) {
+        Ok(val) => env_var_val = val,
+        Err(_e) => env_var_val = "NoDaemon".to_string(),
+    }
+
+    if env_var_val.to_lowercase() == "false" || env_var_val.to_lowercase() == "no" {
```

Comment:
Should we also support `0` as a falsy value? I don't think any change is required here, just curious 🙂 I'm not sure what recommended practices are for allowed values to specify a boolean flag in an environment variable.

Reply:
When we doc this, I suggest that we only document "no". I added the "false" option for those who refuse to read doc. In that spirit, adding 0 is also reasonable. I added that third option.